Core/src/org/sleuthkit/autopsy/datamodel/BlackboardArtifactNode.java

```diff
@@ -124,8 +123,20 @@ public BlackboardArtifactNode(BlackboardArtifact artifact, String iconPath) {
         super(Children.LEAF, createLookup(artifact));
 
         this.artifact = artifact;
-        //this.associated = getAssociatedContent(artifact);
-        this.associated = this.getLookup().lookup(Content.class);
+        
+        // Look for associated Content  i.e. the source file for the artifact
+        if (this.getLookup().lookupAll(Content.class).size() > 1) {
+            for (Content content : this.getLookup().lookupAll(Content.class)) {
+                if ( (content != null)  && (!(content instanceof BlackboardArtifact)) ){
+                    this.associated = content;
+                    break;
+                }
+            }
+        }
+        if (null == this.associated ) {
+            this.associated = this.getLookup().lookup(Content.class);
+        }
+        
```

Comment:
I am a bit confused by this code. I don't understand why we need to call lookupAll twice - won't the for loop just not execute if the collection returned is empty? Also, if we do not find associated via lookupAll(Content.class), why would we expect to find it via lookup(Content.class)? 

If we are going to have several lines of duplicated code here. it would be nice to have a littlr private helper method for this, called by both constructors.